tests/CommunityToolkit.Mvvm.UnitTests/Test_ICommandAttribute.cs

```diff
@@ -567,4 +616,45 @@ public sealed class User
     {
         public string? Name { get; set; }
     }
+
+    public partial class CancelCommandViewModel
+    {
+        public TaskCompletionSource<object?> Tcs1 { get; } = new();
+
+        public TaskCompletionSource<object?> Tcs2 { get; } = new();
+
+        [ICommand(IncludeCancelCommand = true)]
+        private async Task DoWorkAsync(CancellationToken token)
+        {
+            using CancellationTokenRegistration registration = token.Register(static state => ((TaskCompletionSource<object?>)state!).TrySetCanceled(), Tcs1);
+
+            try
+            {
+                _ = await Tcs1.Task;
+
+                _ = Tcs1.TrySetResult(null);
+            }
+            catch (OperationCanceledException e)
+            {
+                _ = Tcs1.TrySetResult(e);
+            }
+        }
+
+        [ICommand(IncludeCancelCommand = true)]
+        private async Task DoWorkWithParameterAsync(int number, CancellationToken token)
+        {
+            using CancellationTokenRegistration registration = token.Register(static state => ((TaskCompletionSource<object?>)state!).TrySetCanceled(), Tcs2);
```

Comment:
Have you made sure this task fails successfully when a nullref happens on the suppressed nullable? I've seen some cases where the MSTest framework doesn't fail properly when an exception throws inside a task, sometimes waiting indefinitely, returning inconclusive, or just passing anyway.

Reply:
That can never be null, because I'm passing that `Tcs2` instance as state, which is not nullable.
It's just that `CancellationToken.Register` doesn't know the state nullability propagates to the state callback.